Microsoft.Toolkit/Collections/ObservableGroup.cs

```diff
@@ -49,9 +52,21 @@ public ObservableGroup(TKey key, IEnumerable<TValue> collection)
         }
 
         /// <summary>
-        /// Gets the key of the group.
+        /// Gets or sets the key of the group.
         /// </summary>
-        public TKey Key { get; }
+        public TKey Key
+        {
+            get => this.key;
+            set
+            {
+                if (!EqualityComparer<TKey>.Default.Equals(this.key, value))
+                {
+                    this.key = value;
+                    
+                    OnPropertyChanged(new PropertyChangedEventArgs(nameof(Key)));
```

Comment:
Perf: since we're always raising the same property changed notification and the `PropertyChangedEventArgs` instances are immutable, could you cache this with a `KeyChangedEventArgs` static field declared at the top of the class? See here (though in this case the field will need to just be private):

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/634e243809b964700d736bbb9bae1f7b8efa464a/Microsoft.Toolkit.Mvvm/Input/AsyncRelayCommand.cs#L22-L25

And then just change this line to `OnPropertyChanged(KeyChangedEventArgs);` 😄

Reply:
Good idea